can/io/mf4.py

```diff
@@ -0,0 +1,468 @@
+            self._mdf.extend(2, sigs)
+
+        elif msg.is_error_frame:
+            if channel is not None:
+                self._err_buffer["CAN_ErrorFrame.BusChannel"] = channel
+
+            self._err_buffer["CAN_ErrorFrame.ID"] = msg.arbitration_id
+            self._err_buffer["CAN_ErrorFrame.IDE"] = int(msg.is_extended_id)
+            self._err_buffer["CAN_ErrorFrame.Dir"] = 0 if msg.is_rx else 1
+            data = msg.data
+            size = len(data)
+            self._err_buffer["CAN_ErrorFrame.DataLength"] = size
+            self._err_buffer["CAN_ErrorFrame.DataBytes"][0, :size] = data
+            if msg.is_fd:
+                self._err_buffer["CAN_ErrorFrame.DLC"] = len2dlc(msg.dlc)
+                self._err_buffer["CAN_ErrorFrame.ESI"] = int(msg.error_state_indicator)
+                self._err_buffer["CAN_ErrorFrame.BRS"] = int(msg.bitrate_switch)
+                self._err_buffer["CAN_ErrorFrame.EDL"] = 1
+            else:
+                self._err_buffer["CAN_ErrorFrame.DLC"] = msg.dlc
+                self._err_buffer["CAN_ErrorFrame.ESI"] = 0
+                self._err_buffer["CAN_ErrorFrame.BRS"] = 0
+                self._err_buffer["CAN_ErrorFrame.EDL"] = 0
+
+            sigs = [(np.array([timestamp]), None), (self._err_buffer, None)]
+            self._mdf.extend(1, sigs)
+
+        else:
+            if channel is not None:
+                self._std_buffer["CAN_DataFrame.BusChannel"] = channel
+
+            self._std_buffer["CAN_DataFrame.ID"] = msg.arbitration_id
+            self._std_buffer["CAN_DataFrame.IDE"] = int(msg.is_extended_id)
+            self._std_buffer["CAN_DataFrame.Dir"] = 0 if msg.is_rx else 1
+            data = msg.data
+            size = len(data)
+            self._std_buffer["CAN_DataFrame.DataLength"] = size
+            self._std_buffer["CAN_DataFrame.DataBytes"][0, :size] = data
+            if msg.is_fd:
+                self._std_buffer["CAN_DataFrame.DLC"] = len2dlc(msg.dlc)
+                self._std_buffer["CAN_DataFrame.ESI"] = int(msg.error_state_indicator)
+                self._std_buffer["CAN_DataFrame.BRS"] = int(msg.bitrate_switch)
+                self._std_buffer["CAN_DataFrame.EDL"] = 1
+            else:
+                self._std_buffer["CAN_DataFrame.DLC"] = msg.dlc
+                self._std_buffer["CAN_DataFrame.ESI"] = 0
+                self._std_buffer["CAN_DataFrame.BRS"] = 0
+                self._std_buffer["CAN_DataFrame.EDL"] = 0
+
+            sigs = [(np.array([timestamp]), None), (self._std_buffer, None)]
+            self._mdf.extend(0, sigs)
+
+        # reset buffer structure
+        self._std_buffer = np.zeros(1, dtype=STD_DTYPE)
+        self._err_buffer = np.zeros(1, dtype=ERR_DTYPE)
+        self._rtr_buffer = np.zeros(1, dtype=RTR_DTYPE)
+
+
+class MF4Reader(MessageReader):
+    """
+    Iterator of CAN messages from a MF4 logging file.
+
+    The MF4Reader only supports MF4 files that were recorded with python-can.
```

Comment:
Why is that?

Reply:
It's a limitation of asammdf. See [here](https://github.com/danielhrisca/asammdf/issues/744#issuecomment-1207942077)